Assets/Scripts/Enemy/AbstractEnemyController.cs

```diff
@@ -5,5 +5,6 @@ namespace Enemy
     public abstract class AbstractEnemyController : MonoBehaviour
     {
         public abstract void TakeDamage(float damage);
+        public abstract float GetCollisionDamage();
```

Comment:
Asta poate să fie abstract property:

```csharp
public abstract float CollisionDamage { get; }
```

[How to define abstract properties - C# Programming Guide](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-define-abstract-properties)